Source/Core/Core/PowerPC/JitArm64/Jit.cpp

```diff
@@ -386,10 +403,32 @@ void JitArm64::WriteExit(u32 destination, bool LK, u32 exit_address_after_return
     linkData.exitPtrs = GetWritableCodePtr();
     linkData.linkStatus = false;
     linkData.call = false;
+    linkData.exitFarcode = farcode_2_addr;
     b->linkData.push_back(linkData);
 
     blocks.WriteLinkBlock(*this, linkData);
   }
+
+  if (switch_to_far_code)
+    SwitchToFarCode();
+  ASSERT(GetCodePtr() == farcode_1_addr || HasWriteFailed());
+  MOVI2R(DISPATCHER_PC, destination);
+  if (LK)
+    BL(GetAsmRoutines()->do_timing);
+  else
+    B(GetAsmRoutines()->do_timing);
+
+  if (LK)
+  {
+    if (GetCodePtr() == farcode_2_addr - sizeof(u32))
+      BRK(101);
```

Comment:
Wouldn't we end up executing that BRK if we return from the BL above?

Also i've noted that in some calls to BRK you're using a hex value and in some a decimal. But i don't think it matters.

Reply:
> Ah the BRK is obviously to fill the 3 instructions so that the return value pushed to the stack is correct. NVM!

Actually, that's not it. Before this PR, `JitArm64BlockCache::WriteLinkBlock` would emit two BLs: The "slow" BL and the "fast" BL. The fast BL matched the address pushed to the stack, but the slow one didn't, so if the slow BL were to be triggered, the BLR optimization would fail. What has happened in this commit is that the slow BL was moved to the farcode you see here. Just like before, it does not match the address pushed to the stack. And because of that, we'll never actually return to here.

The actual reason why we need this is because we made a "prediction" earlier that `farcode_size` would be 3 instructions long. If it turned out to be 2 instructions long (`MOVI2R` can emit either 1 or 2 instructions in this situation), we just pad it so it's 3 instructions long. It wouldn't be impossible to make `farcode_size` correctly predict whether we get 2 or 3 instructions, but it seems unnecessarily complicated to me.